tests/installation/upgrade_select.pm

```diff
@@ -39,7 +39,10 @@ sub run {
         send_key $cmd{next};
     }
     if (match_has_tag("select-for-update")) {
-        send_key $cmd{next};
+        assert_screen('select-for-update-arch');
```

Comment:
I suggest using $arch as tag, $arch=setting(platform), then we check specific screen for specific platform.

Reply:
Thanks for comment, I changed the code as your suggestion.